mailpoet/lib/Validator/Schema/ObjectSchema.php

```diff
@@ -0,0 +1,55 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Validator\Schema;
+
+use MailPoet\Validator\Schema;
+
+// See: https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/#objects
+class ObjectSchema extends Schema {
+  protected $schema = [
+    'type' => 'object',
+  ];
+
+  /** @param array<string, Schema> $properties */
+  public function properties(array $properties): self {
```

Comment:
We can do that, I was probably thinking that since there are also pattern and additional properties, I'd just leave a method for that. I was also thinking about different APIs for making required/optional properties easier to write.

Reply:
Looking at it... if I add the constructor, I should probably add one to the array schema as well. Maybe it's fine as it is, considering `Builder` will be used anyway? The difference from `oneOf`/`anyOf` is they don't really make much sense without subschemas while types like object and array can be used without property and item definitions.

Reply:
Very good point about empty objects/arrays being valid! And `oneOf`/`anyOf` do feel like special cases.

On that note, you are allowing `oneOf`/`anyOf` to be instantiated with an empty array of schemas while the `Builder` doesn't allow for that. Should those things be made consistent? Probably not important if the Builder is the main interface for using those schemas, but it does seem a little strange to allow schemas that don't make much sense.

Reply:
Good point, I don't know why I had the default values for `oneOf`/`anyOf` in the first place. I removed the defaults.